internal/envoy/route.go

```diff
@@ -135,33 +135,33 @@ func responseTimeout(r *dag.Route) *duration.Duration {
 	if r.TimeoutPolicy == nil {
 		return nil
 	}
-	return timeout(r.TimeoutPolicy.ResponseTimeout)
+	return envoyTimeout(r.TimeoutPolicy.ResponseTimeout)
 }
 
 func idleTimeout(r *dag.Route) *duration.Duration {
```

Comment:
You can clean this up further by making the `dag.Route.TimeoutPolicy` a non-pointer member that defaults all its fields to `setting.DefaultSetting()`. Then you would  not need these wrappers just to fetch specific fields, you could just do `envoyTimeout(r.TimeoutPolicy.ResponseTimeout)` directly in the caller.